smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -81,8 +81,9 @@ contract MixinTransfer is
       emit ExpireKey(_tokenId);
     }
 
-    if (toKey.tokenId == 0) {
```

Comment:
hmm i'm not sure i understand what's going on here.  how is using iDto better than toKey.tokenId, on line 62 we set iDTo to the toKey.tokenId

Reply:
Fair point. This change is just on optimization, it saves 1 SLOAD by using the cached value. The fix is on line 86, where we update iDTo with the newly assigned tokenId.

Reply:
ah i see